source/s3_meta_request.c

```diff
@@ -1071,28 +1071,41 @@ void aws_s3_meta_request_send_request(struct aws_s3_meta_request *meta_request,
 
     AWS_LOGF_TRACE(AWS_LS_S3_META_REQUEST, "id=%p: Sending request %p", (void *)meta_request, (void *)request);
 
-    if (aws_http_stream_activate(stream) != AWS_OP_SUCCESS) {
-        aws_http_stream_release(stream);
-        stream = NULL;
-
-        AWS_LOGF_ERROR(
-            AWS_LS_S3_META_REQUEST, "id=%p: Could not activate HTTP stream %p", (void *)meta_request, (void *)request);
-
-        goto error_finish;
-    }
-
     {
         /* BEGIN CRITICAL SECTION */
         aws_s3_meta_request_lock_synced_data(meta_request);
+        if (aws_s3_meta_request_has_finish_result_synced(meta_request) && !request->always_send) {
+            /* The meta request is finishing up, drop the HTTP stream to the ground. */
+            aws_raise_error(meta_request->synced_data.finish_result.error_code);
```

Comment:
trivial: should we raise some kind of CANCELED error here instead? It's not going to end up as the final meta-request error code, because it already has one. But it might make more sense in the logs to see parts getting torn down with CANCELED errors